cmd/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+type Server struct {
+	mux *chi.Mux
+}
+
+func main() {
+	srv := Server{
+		mux: chi.NewRouter(),
+	}
+	srv.mux.Use(middleware.Logger)
+	srv.mux.Use(middleware.Recoverer)
+	srv.mux.Get("/hello-world", HelloWorldHandler)
+
+	srv.mux.Post("/todo", TodoHandler)
+
+	srv.mux.Get("/healthz", HealthzHandler)
+
+	log.Fatal(http.ListenAndServe(":8080", srv.mux))
+}
+
+func TodoHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could add this to your server struct, then you can do srv.TodoHandler to register it. And by putting it on your struct, if in the future you have a database, your handler can then access it. 

```go
type Server struct {
    mux *chi.Mux
    db Store // Store would very likely be an interface, have a play to see what I mean :)  
}

func (s *Server) TodoHandler(w http.ResponseWriter, r *http.Request) {
    todos, err := s.db.GetTodos(userId)


...rest of code
```